geode-core/src/main/java/org/apache/geode/management/internal/api/LocatorClusterManagementService.java

```diff
@@ -468,11 +467,30 @@ public LocatorClusterManagementService(ConfigurationPersistenceService persisten
     return result;
   }
 
-  @Override
+  private <T extends ClusterManagementResult> T assertSuccessful(T result) {
```

Comment:
In the case where we know we are being called from the REST API, we need to return the unsuccessful cluster management result rather than throw exception (to avoid losing rich failure info).

Reply:
fixed by adding controller advice to catch our own exception and unwrap it